docs/internals/security.rst

```diff
@@ -223,14 +223,17 @@ Depending on the chosen mode (see :ref:`borg_init`) different primitives are use
   and is also tracked locally on the client to avoid counter reuse.
 
 - The authentication primitive is either HMAC-SHA-256 or BLAKE2b-256
-  in a keyed mode. HMAC-SHA-256 uses 256 bit keys, while BLAKE2b-256
-  uses 512 bit keys.
-
-  The latter is secure not only because BLAKE2b itself is not
-  susceptible to `length extension`_, but also since it truncates the
-  hash output from 512 bits to 256 bits, which would make the
-  construction safe even if BLAKE2b were broken regarding length
-  extension or similar attacks.
+  in a keyed mode. 
+
+  Both HMAC-SHA-256 and BLAKE2b have undergone extensive cryptanalysis  
+  and have proven secure against known attacks. The known vulnerability
+  of SHA-256 against length extension attacks does not apply to HMAC-SHA-256.
+  
+  The authentication primitive should be chosen based upon SHA hardware support:
+  AMD Ryzen and new 12th+ Generation Intel Core processors (beginning with 
```

Comment:
I checked the CPU support and according to these sources, we should change this a bit:

https://en.wikichip.org/wiki/intel/microarchitectures/ice_lake_(client) -> "10th+ generation / Ice Lake"

https://en.wikichip.org/wiki/amd/microarchitectures/zen -> "Zen/Ryzen/EPYC"

Reply:
Note that ice lake was limited to mobile, so it's not just the generation number, but also which kind of SKU, which determines uarch and thus ISA extensions. For desktops 12th+ is correct.

Reply:
*actually it seems rocket lake already included it - so 10th+ for mobile, 11th+ for desktop should be right.

Reply:
updated.